services/question-service/entry-points/api/server.ts

```diff
@@ -1,19 +1,33 @@
-import { Server } from 'http';
-import { AddressInfo } from 'net';
-import express from 'express';
-import defineRoutes from './routes';
-import { errorHandler } from '../../errorHandler';
-const cors = require('cors');
+import express from "express";
+import type { Server } from "http";
+import type { AddressInfo } from "net";
+
+import { errorHandler } from "../../errorHandler";
+
+import defineRoutes from "./routes";
+
+const cors = require("cors");
 
 let connection: Server;
 
+async function openConnection(
+  expressApp: express.Application,
+): Promise<AddressInfo> {
+  return new Promise((resolve) => {
+    const webServerPort = process.env.PORT || 0;
```

Comment:
I think we can set a valid port as a default value, perhaps like `5001` or something?

Reply:
Right, good point, have updated in my latest commit. Thank you for your reviews!

Reply:
No worries! Also just a small note but I think you should use perhaps ports from `1001`-`9001`. Although they run on different servers, `3000` is often used for the frontend, so I feel that we should avoid it to reduce confusion haha.

Reply:
Okie sure, will take note haha. I had updated it to 5001 and not 3000 so the changes should still be good :D